packages/story-editor/src/components/panels/document/taxonomies/FlatTermSelector.js

```diff
@@ -89,11 +91,21 @@ function FlatTermSelector({ taxonomy, canCreateTerms }) {
     [canCreateTerms, terms, taxonomy, termCache, setTerms, createTerm]
   );
 
-  const handleFreeformInputChange = useDebouncedCallback((value) => {
+  const handleFreeformInputChange = useDebouncedCallback(async (value) => {
+    if (value.length === 0) {
+      setSearchResults([]);
+      return;
+    }
+
     if (value.length < 3) {
       return;
     }
-    addSearchResultsToCache(taxonomy, { name: value });
+    const results = await addSearchResultsToCache(taxonomy, {
+      search: value,
+      // This is the per_page value Gutenberg is using
+      per_page: 20,
+    });
+    setSearchResults(results);
   }, 1000);
```

Comment:
🤔 Do we want to wait a whole second here? Feels like there's a lot of lag. maybe 300 ms would feel a little more natural and still not shoot a request off each keystroke.

<img src="https://user-images.githubusercontent.com/22185279/135681053-0e85214e-e927-4a31-bb0d-ef9292f64cd1.gif" height="300px" />

Reply:
does feel like a lot, this was existing so i had just left it. i'll reduce it to 300

Reply:
yea I just put it in there arbitrarily.  reducing to 300 works for me